src/datasets/utils/metadata.py

```diff
@@ -106,6 +111,68 @@ def to_yaml_string(self) -> str:
         ).decode("utf-8")
 
 
+class MetadataConfigs(Dict[str, dict]):
+    """Should be in format {config_name: {**config_params}}."""
+
+    __configs_field_name: ClassVar[str] = METADATA_CONFIGS_FIELD
+
+    @classmethod
+    def from_metadata(cls, dataset_metadata: DatasetMetadata) -> "MetadataConfigs":
+        if cls.__configs_field_name in dataset_metadata:
+            metadata_configs = dataset_metadata.get(cls.__configs_field_name)
+            if isinstance(metadata_configs, dict):
+                config_name = metadata_configs.pop("config_name", "default")
+                return cls({config_name: metadata_configs})
+            if isinstance(metadata_configs, list):
+                return cls(
+                    {
+                        config["config_name"]: {
+                            param: value for param, value in config.items() if param != "config_name"
+                        }
+                        for config in metadata_configs
+                    }
+                )
+        return cls()
```

Comment:
I was wondering if this implementation could be made simpler by reassigning:
```python
metadata_configs = [metadata_configs] if isinstance(metadata_configs, dict) else metadata_configs
```
What do you think?